symfit/core/fit.py

```diff
@@ -1235,93 +1027,147 @@ def execute(self, relative_error=1e-8, max_iter=500):
-    and :class:`ConstrainedNumericalLeastSquares`.
-    """
-    def __init__(self, *args, **kwargs):
-        if 'constraints' in kwargs:
-            self.fit = ConstrainedNumericalLeastSquares(*args, **kwargs)
-        else:
-            init = TakesData(*args, **kwargs)
-            # self.model = init.model
-            if not all(min is None and max is None for min, max in init.model.bounds):
-                # Bounds have been set.
-                self.fit = ConstrainedNumericalLeastSquares(*args, **kwargs)
-            else:
-                if init.model.shared_parameters:
-                    self.fit = ConstrainedNumericalLeastSquares(*args, **kwargs)
-                else:
-                    self.fit = NumericalLeastSquares(*args, **kwargs)
+    Example usage::
 
-    def execute(self, *options, **kwoptions):
-        """
-        Execute ``Fit``, giving any ``options`` and ``kwoptions`` to the
-        fitting object.
-        """
-        return self.fit.execute(*options, **kwoptions)
+        a, b = parameters('a, b')
+        x, y = variables('x, y')
 
-    @property
-    def model(self):
-        """
-        Property which returns the :class:`Model` object, taken from
-        self.fit.model
-        """
-        return self.fit.model
+        model = {y: a * x + b}
 
-    @model.setter
-    def model(self, new_model):
-        self.fit.model = new_model
+        # Fit will use its default settings
+        fit = Fit(model, x=xdata, y=ydata)
+        fit_result = fit.execute()
 
+        # Use Nelder-Mead instead
+        fit = Fit(model, x=xdata, y=ydata, minimizer=NelderMead)
+        fit_result = fit.execute()
 
-class Minimize(BaseFit):
     """
-    Minimize a model subject to constraints. A wrapper for ``scipy.optimize.minimize``.
 
-    ``Minimize`` itself doesn't work when data is provided to its Variables, use
-    one of its subclasses for that.
-    """
-    @keywordonly(constraints=None)
-    def __init__(self, model, *args, **kwargs):
+    @keywordonly(objective=None, minimizer=None, constraints=None)
+    def __init__(self, model, *ordered_data, **named_data):
         """
-        Because in a lot of use cases for Minimize no data is supplied to variables,
-        all the empty variables are replaced by an empty np array.
 
-        :constraints: constraints the minimization is subject to.
-        :type constraints: list
+        :param model: (dict of) sympy expression(s) or ``Model`` object.
+        :param constraints: iterable of ``Relation`` objects to be used as
+            constraints.
+        :param bool absolute_sigma: True by default. If the sigma is only used
+            for relative weights in your problem, you could consider setting it to
+            False, but if your sigma are measurement errors, keep it at True.
+            Note that curve_fit has this set to False by default, which is wrong in
+            experimental science.
```

Comment:
`absolute_sigma` is kicked up to `TakesData`, who handles it. The keyword_only decorator is such that if you inspect this `__init__` it will actually contain `absolute_sigma` as well, and so it is sphynx friendly ;).